jni_bindings/src/lib.rs

```diff
@@ -278,176 +341,192 @@ fn from_attr(attr: &dom::Attribute) -> Attribute {
-    let result = catch_unwind(|| unsafe {
-        let element: &Element = jlong_to_pointer::<Element>(element).as_mut().unwrap();
-        from_std_string_jstring(String::from(element.namespace.to_str()), env)
-    });
-
-    match result {
-        Ok(jstring) => jstring,
-        Err(err) => {
-            let exception_class = clone.find_class("java/lang/RuntimeException").unwrap();
-            let message = format!(
-                "Rust panic occurred in get_node_element_namespace: {:?}",
-                err
-            );
-            clone.throw_new(exception_class, message).unwrap();
-            clone.new_string("").unwrap().into()
-        }
+    let this = this as *mut Element;
+
+    if this.is_null() {
+        env.throw_new(
+            "java/lang/NullPointerException",
+            "Document::get_node_element_namespace called with null pointer",
+        )
+        .unwrap();
+
+        JObject::null().into()
+    } else {
+        let element = &*this;
+        env.new_string(element.namespace.to_str())
+            .unwrap_or_else(|_| JObject::null().into())
     }
 }
 
 #[no_mangle]
-pub extern "system" fn Java_org_phoenixframework_liveview_lib_Document_get_1node_1element_1tag<
+pub unsafe extern "system" fn Java_org_phoenixframework_liveview_lib_Node_00024Element_get_1tag<
     'local,
 >(
-    env: JNIEnv<'local>,
+    mut env: JNIEnv<'local>,
     _: JClass<'local>,
-    element: jlong,
+    // non-null raw pointer to Element
+    this: jlong,
 ) -> JString<'local> {
-    let mut clone = unsafe { env.unsafe_clone() };
-    let result = catch_unwind(|| {
-        let element: &Element = unsafe { jlong_to_pointer::<Element>(element).as_mut().unwrap() };
-        from_std_string_jstring(String::from(element.tag.to_str()), env)
-    });
-
-    match result {
-        Ok(jstring) => jstring,
-        Err(err) => {
-            let exception_class = clone.find_class("java/lang/RuntimeException").unwrap();
-            let message = format!("Rust panic occurred in get_node_element_tag: {:?}", err);
-            clone.throw_new(exception_class, message).unwrap();
-            clone.new_string("").unwrap().into()
-        }
+    let this = this as *mut Element;
+
+    if this.is_null() {
+        env.throw_new(
+            "java/lang/NullPointerException",
+            "Document::get_node_element_tag called with null pointer",
+        )
+        .unwrap();
+
+        JObject::null().into()
+    } else {
+        let element = &*this;
+        env.new_string(element.tag.to_str())
+            .unwrap_or_else(|_| JObject::null().into())
     }
 }
 
 #[no_mangle]
-pub extern "system" fn Java_org_phoenixframework_liveview_lib_Document_get_1node_1element_1attributes<
+pub unsafe extern "system" fn Java_org_phoenixframework_liveview_lib_Node_00024Element_get_1attributes<
     'local,
 >(
-    env: JNIEnv<'local>,
+    mut env: JNIEnv<'local>,
     _: JClass<'local>,
-    element: jlong,
+    // non-null raw pointer to Element
```

Comment:
The `non-null` can't be true or there would be no need for the `this.is_null()` check below.
```suggestion
    // raw pointer to Element
```